internal/retry/retry.go

```diff
@@ -74,10 +74,23 @@ func (r *Retryer) runRetryLoop(
 	sleepTime := minSleepTime
 
 	for {
+		if beforeFunc, hasBefore := r.before.Get(); hasBefore {
+			beforeFunc()
+		}
+
 		eg, egCtx := errgroup.WithContext(ctx)
 		for i, curFunc := range funcs {
+			if curFunc == nil {
+				panic("curFunc should be non-nil")
+			}
 
 			eg.Go(func() error {
+				if curFunc == nil {
```

Comment:
Why do we check curFunc to be non-nil both twice?

Reply:
This 2nd check is probably redundant. I was just trying to suss out why I was seeing segfaults, and I figured I might as well leave the panics in.